aten/src/ATen/native/mps/operations/TensorCompare.mm

```diff
@@ -268,6 +269,52 @@ static void clamp_scalar_out_mps(const Tensor& input_t,
   }
 }
 
+static void isin_Tensor_Tensor_out_mps(const Tensor& elements,
+                                       const Tensor& test_elements,
+                                       bool assume_unique,
+                                       bool invert,
+                                       const Tensor& out,
+                                       string op_name) {
+  if (elements.numel() == 0) {
```

Comment:
Please add `TORCH_CHECK(elements.is_mps() && test_elements.is_mps())` as well as `TORCH_CHECK(elements.dtype() == test_elements.dtype())`

Reply:
Added the checks and tests for them.